### PR TITLE
Highlighting via comment

### PR DESCRIPTION
**Pull request by fnpen**
@julienp I've prepared changes.

**Any:**
```js
const Component = /* styled */`
  display: block;
`;
```

**Next.js:**
```js
const Component = Component.styled/* styled */`
  display: block;
`;
```

**Custom:**
```js
const Component = styledWithMyCustomProcessor/* styled */`
  display: block;
`;
```

`/* styled */` is comfortable way for any cases like manual usage:
```js
class Component {
    styled = /* styled */`
        display: block;
    `;

    getComponent() {
        return styled(this.styled);
    }
}
```

### CODE REVIEW

File: syntaxes/styled-components.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"fileTypes": ["js", "jsx", "ts", "tsx"],
-	"injectionSelector": "L:source -comment",
+	"injectionSelector": "L:source",
```
Comment (Cu3PO42) [inline syntaxes/styled-components.json]: This causes CSS to be highlighted inside of comments which is not the desired behaviour I believe.

It should also not be necessary for your change to work.

Comment (fnpen) [reply syntaxes/styled-components.json]: @Cu3PO42  I've tried with it. ` -comment` exclude comments and RegExp for comment not working.

Comment (Cu3PO42) [reply syntaxes/styled-components.json]: `-comment` does not prevent you from consuming comments, it prevents the rule from being injected into the scope of a comment.

I just tried your branch with `-comment` and it works as expected.

**Comment by sampotts** [comment]
+1 to this. I'd like to use this for mixins

**Comment by juanpasolano** [comment]
I'd love this feature too!

**Comment by hmil** [comment]
How about a having the comment inside the template, like
```js
const Component = Component.styled`
  /* styled */
  display: block;
`;
```

or even something more descriptive like:
```js
const Component = Component.styled`
  /* type:style */
  display: block;
`;
```

instead of

```js
const Component = Component.styled/* styled */`
  display: block;
`;
```

It's just a suggestion because I like the idea of using a comment but I feel like the syntax you suggest looks awkward. This looks more like linter instruction comments which people are more familiar with.

**Comment by taylon** [comment]
Is there anything missing for getting this merged?
I'm happy to help if that is the case.

**Comment by targetcreature** [comment]
+1, would love to see this merged. Thanks all for your hard work!

**Comment by Nick-Lucas** [comment]
+2, this would solve a ton of open issues, and make it easy to write factory functions for style string fragments, without giving up intellisense support

**Comment by kevinrodriguez-io** [comment]
When are we going to ship this?

**Comment by jasonwilliams** [comment]
@fnpen could you rebase your changes?

**Comment by a7madgamal** [comment]
@fnpen rebase PLZZ :D
image

**Comment by jasonwilliams** [comment]
Im closing this PR due to no activity from the original committer

**Pull request closed without merging**